File: documind-frontend/services/userService.ts
import api from "./api";

import type { User } from "../types/user";



export async function getCurrentUser(): Promise<User> {

    const response = await api.get<User>(
        "/users/me"
    );

    return response.data;

}




export async function updateProfile(

    data: {
        firstname: string;
        lastname: string;
    }

): Promise<User> {


    const response = await api.put<User>(
        "/users/me",
        data
    );


    return response.data;

}

File: documind-backend/app/services/user_service.py
from sqlalchemy.orm import Session

from app.repositories.user_repository import UserRepository

from app.schemas.user import UserUpdate



class UserService:


    def __init__(self, db: Session):

        self.repository = UserRepository(db)



    def get_user_by_id(
        self,
        user_id: int
    ):

        user = self.repository.get_by_id(user_id)


        if not user:

            raise Exception(
                "User not found"
            )


        return user





    def update_profile(
        self,
        user_id: int,
        data: UserUpdate
    ):


        user = self.repository.get_by_id(
            user_id
        )


        if not user:

            raise Exception(
                "User not found"
            )



        user.firstname = data.firstname

        user.lastname = data.lastname



        return self.repository.update(
            user
        )